src/reducers.js

```diff
@@ -0,0 +1,50 @@
+import calculateWinner from './winner';
+
+const initialState = {
+  history: [Array(9).fill(null)],
+  xIsNext: true,
+  stepNumber: 0,
+  winner: null
+};
+
+const play = (state, action) => {
+  const { history, xIsNext } = state;
+  const latestPlay = history[history.length - 1];
+  if (latestPlay[action.pos] || calculateWinner(latestPlay)) {
```

Comment:
I don't quite follow this logic... sounds a bit convoluted to me. I would probably have a function called `isGameFinished` and/or `isSquareAvaliable(pos)` or something like that and if those tell the move is invalid simply return `state`. Then have a more straightforward push of the new move

Reply:
I agree it's not be-a-u-tiful 💅 nor very readable 👓 

If valid, my disclaimer is that I was not paying attention to "business/game logic" to focus on react/redux learning. Many things come right out from the react tutorial. I should have warned you before dragging you into this review!